platforms/ios/src/TangramMap/TGMapViewController.mm

```diff
@@ -187,49 +199,41 @@ - (void)loadSceneFileAsync:(NSString *)path sceneUpdates:(NSArray<TGSceneUpdate
         }
     }
 
-    auto updateCallbackStatus = [=](auto sceneUpdateError) {
-        if (!self.mapViewDelegate || ![self.mapViewDelegate respondsToSelector:@selector(mapView:didFailSceneUpdateWithError:)]) { return; }
-        [self.mapViewDelegate mapView:self didFailSceneUpdateWithError:[TGHelpers errorFromSceneUpdateError:sceneUpdateError]];
-    };
-
-    self.map->loadSceneAsync([path UTF8String], false, onReadyCallback, nullptr, updates, updateCallbackStatus);
+    self.map->setSceneReadyListener(onReadyCallback);
```

Comment:
We could define this for convenience:
```
self.map->setSceneReadyListener([self sceneReadyListener]);
```
With a method defined as such
```
- (Tangram::SceneReadyCallback) sceneReadyListener {
    auto listener = [=](int sceneID, auto sceneError) {
        // ...
    }
    return listener;
}
```
And use it on line 235, and 166 too so we don't miss marker removal and render request from the listener.